arm-visualstudio/2014-04-01-preview/swagger/Csm.json

```diff
@@ -0,0 +1,881 @@
+          "accountResource"
+        ],
+        "summary": "Accounts_Get",
+        "description": "Gets the Visual Studio Team Services account resource details.",
+        "operationId": "accounts_Get",
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "$ref": "#/parameters/resourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/subscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          },
+          {
+            "in": "path",
+            "name": "resourceName",
+            "description": "Name of the resource.",
+            "required": true,
+            "type": "string"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The operation succeeded. The response contains the details of the Visual Studio Team Services account resource.",
+            "schema": {
+              "$ref": "#/definitions/AccountResourceGetResponse"
+            }
+         },
+          "404": {
+            "description": "The Visual Studio Team Services account does not exist."
+         }
+        },
+        "x-ms-examples": {
+            "getAccountResource": {
+                "$ref": "../examples/GetResource.json"
+            }
+        }   	
+      }
+    }
+  },
+  "definitions": {
+    "AccountResourceGetResponse": {
+      "description": "The response to an account resource GET request.",
+      "type": "object",
+      "properties": {
+        "id": {
+          "description": "Unique identifier of the resource.",
+          "readOnly": true,
+          "type": "string"
+        },
+        "location": {
+          "description": "Resource location.",
+          "type": "string"
+        },
+        "name": {
+          "description": "Resource name.",
+          "readOnly": true,
+          "type": "string"
+        },
+        "properties": {
+          "description": "Resource properties.",
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          },
+          "x-ms-client-flatten": true
+        },
+        "tags": {
+          "description": "Resource tags.",
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          }
+        },
+        "type": {
+          "description": "Resource type.",
+          "readOnly": true,
+          "type": "string"
+        }
+      },
+      "x-ms-azure-resource": true
+    },
+    "AccountResourceGroupGetResponse": {
+      "description": "The response to an account resource group GET request.",
+      "type": "object",
+      "properties": {
+        "value": {
+          "description": "Array of resource details.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/AccountResourceGetResponse"
+          }
+        }
+      }
+    },
+    "AccountResourcePutRequestData": {
```

Comment:
1. I changed our tooling to mark any properties without a public setter as readOnly. This is just an implementation detail on our end so I will make sure this doesn't end up in the Swagger.

2. Same as above.

3. There are 3 different operation types possible that drastically alter the behavior of this operation. `Create` is used for new accounts, `Update` is used for existing accounts, and `Link` links the account to an Azure subscription in VSTS Commerce.

This API has existed for quite some time (hence the 2014 API version) and would be difficult to modify at this point, otherwise I would have suggested to the team that we make `Update` a PATCH operation, and `Link` could also be a separate option. Would be no need for `operationType` in that case.